### PR TITLE
DAOS-12114 build: remove psm2 dependency

### PR DESCRIPTION
**Pull request by soumagne**
Remove also PSM2 support from CART and psm2 references in tests

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by frostedcmos — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
Looks like this doc need an update:
```
daos/docs/dev/development.md
```
And maybe these?
```
daos/src/control/cmd/daos_server/auto_test.go
daos/src/control/cmd/dmg/auto_test.go
daos/src/control/lib/control/auto_test.go
daos/src/control/lib/control/mocks.go
daos/src/control/lib/control/mocks.go
daos/src/control/lib/hardware/libfabric/provider_test.go
```

File: src/cart/README.env [hunk 1]
```diff
@@ -2,14 +2,13 @@ This file lists the environment variables used in CaRT.
...
  . D_PROVIDER (Deprecated: CRT_PHY_ADDR_STR)
    It determines which mercury NA plugin to be used:
-   1: set it as "ofi+verbs;ofi_rxm" to use OFI verbs;ofi_rxm provider
-   2: set it as "ofi+psm2" to use OFI psm2 provider
-   3: set it as "ofi+gni" to use OFI gni provider
-   4: set it as "sm" to use SM plugin which only works within single node
-   5: set it as "ofi+tcp;ofi_rxm" to use OFI tcp;ofi_rxm provider.
-   6: set it as "ofi+sockets" to use OFI sockets provider
+   - set it as "ofi+verbs;ofi_rxm" to use OFI verbs;ofi_rxm provider
+   - set it as "ofi+gni" to use OFI gni provider
```
Comment (frostedcmos) [inline src/cart/README.env:6]: we should remove ofi+gni references too, i dont even know what gni provider is :)

**Review by frostedcmos — COMMENTED** [review]
few inline comments

**Comment by soumagne** [comment]
@daltonbohning I think I would appreciate it if someone else from the control plane team (@kjacque please feel free to chime in) or someone else could remove the psm2 references from the go tests since those tests might need to be reworked a bit more. Maybe that can be a separate PR ?

File: site_scons/prereq_tools/base.py [hunk 2]
```diff
@@ -43,8 +43,6 @@
 from SCons.Script import BUILD_TARGETS
 from SCons.Errors import InternalError
 
-OPTIONAL_COMPS = ['psm2']
```
[the remark this answers is not in the document]
Comment (soumagne) [reply site_scons/prereq_tools/base.py:46]: @jolivier23 @ashleypittman any preferences ? should I keep `OPTIONAL_COMPS = ['']` ?

File: src/cart/README.env (hunk 1 above)
Comment (soumagne) [reply src/cart/README.env:6]: Yes sure I was not trying to rewrite the readme as part of that PR but if I go that route we should make it match the crt type list yeah or refer to the doc if there is any doc.

File: site_scons/prereq_tools/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jolivier23) [reply site_scons/prereq_tools/base.py:46]: Good question.   I'd be fine removing it.

**Comment by daltonbohning** [comment]
> @daltonbohning I think I would appreciate it if someone else from the control plane team (@kjacque please feel free to chime in) or someone else could remove the psm2 references from the go tests since those tests might need to be reworked a bit more. Maybe that can be a separate PR ?

Sure, no problem from me. I don't know the answer either :)

**Review by jolivier23 — DISMISSED** [review]

**Comment by kjacque** [comment]
>  I think I would appreciate it if someone else from the control plane team (@kjacque please feel free to chime in) or someone else could remove the psm2 references from the go tests since those tests might need to be reworked a bit more. Maybe that can be a separate PR ?

I don't think there's any work that's required to be done in the Go tests. In general those are unit tests, and none of them interact with the real psm2 provider or Cart.

**Comment by soumagne** [comment]
NB. I won't fix the clang format in that PR to prevent any further conflicts. We should run clang-format on the cart code in a separate PR.

File: src/cart/README.env (hunk 1 above)
Comment (soumagne) [reply src/cart/README.env:6]: I will leave the ofi+gni reference for now. I'd prefer to focus that PR on psm2 and not also remove the gni one from the cart code, we can do that later.

File: site_scons/prereq_tools/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (soumagne) [reply site_scons/prereq_tools/base.py:46]: ok I kept it removed and I also removed the doc section about it.

**Review by frostedcmos — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Pull request merged**